CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+
+project(az_iot_middleware_freertos LANGUAGES C)
+
+set(CMAKE_C_STANDARD 99)
+
+# Add modules and include compiler options/switches
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake-modules")
+# include(eng/cmake/global_compile_options.txt)
+
+# Options for building the lib
+option(build_tests "Build the tests for the Azure IoT Middleware for FreeRTOS" OFF)
+set(config_directory CACHE STRING "The directory which has the FreeRTOSConfig.h and azure_iot_config.h")
+
+# If samples and tests aren't being built, then the user needs to provide a directory which has the necessary configs.
+if(NOT build_tests AND "${config_directory}" STREQUAL "")
+  message(FATAL_ERROR "The library needs a FreeRTOSConfig.h and azure_iot_config.h directory. Please set the path to the directory with those files.")
+endif()
+
+# If testing, turn on relevant switches and add configurations.
+if(build_tests)
+  include(CTest)
+  enable_testing()
+
+  # Turn on the flags necessary for code coverage
+  add_compile_options(-fprofile-arcs -ftest-coverage -O0 -g3)
+  add_link_options(-fprofile-arcs -ftest-coverage)
+
+  # Set the port and config files
+  set(AZURE_IOT_MQTT_PORT ${CMAKE_CURRENT_LIST_DIR}/tests)
```

Comment:
is there a convention to name variable (capital like marco or camel case).

Reply:
Yup usually capitals snake case